### ISSUE TITLE
Speed up renewal

### ISSUE THREAD

**Issue by seabbs** [epiforecasts/EpiSoon#87]
The renewal code can be sped up using a dot product vs its current implementation. This a core part of the simulation code and so would give a good speed up. However, if planning on using thee `EpiNow2` simulation model then this change is not needed as this part of the code base will be depreciated I would imagine. 

https://github.com/epiforecasts/EpiSoon/blob/d8a55315182e26b9b46bdd36e62d44d2da097709/R/predict.R#L111

**Comment by sbfnk** [comment]
Yes, making `EpiNow2` a dependency and fully basing it on that probably makes sense.

**Comment by seabbs** [comment]
The cost is of course the terrifying requirement for `stan`. There could be an easy argument for having a duplicate R based simulator in this package (which wouldn't need to have feature parity) and making it optional for the user. 

The dependency structure would get complicated but that could also have utility in `EpiNow2` (this was how `EpiNow` and the forecasting component of `EpiNow2` worked before being depreciated.
